### PR TITLE
Eval harness 

### PR DESCRIPTION
**Pull request by HamidShojanazeri**
# What does this PR do?
Adding Eval harness from EleutherAI. 

-->

Fixes # (issue)

## Feature/Issue validation/testing

Please describe the tests that you ran to verify your changes and relevant result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
[eval-test-logs ](https://gist.github.com/HamidShojanazeri/d492c10d5bf579d3229c56103f1e00b4)
[accelerate-logs-parallelize](https://gist.github.com/HamidShojanazeri/63224baf901b84d50d317a4c2745fb47)
[vllm test logs](https://gist.github.com/HamidShojanazeri/9d36d95b19fea39598cfa989cd24b796)

- [ ] Test B
Logs for Test B

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/facebookresearch/llama-recipes/blob/main/CONTRIBUTING.md#pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes?  
- [ ] Did you write any new necessary tests?

Thanks for contributing 🎉!

### CODE REVIEW

File: eval/eval.py [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+import logging
+import os
+import re
+import sys
+from pathlib import Path
+
+import numpy as np
+import lm_eval
+from lm_eval import evaluator, tasks
+from lm_eval.utils import make_table, load_yaml_config
```
Comment (mreso) [inline eval/eval.py:12]: unused import: load_yaml_config

File: eval/open_llm_leaderboard/hellaswag_utils.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import datasets
```
Comment (mreso) [inline eval/open_llm_leaderboard/hellaswag_utils.py:1]: This script is also part for the lm_eval repo https://github.com/EleutherAI/lm-evaluation-harness/blob/main/lm_eval/tasks/hellaswag/utils.py
Is this copy actually used and necessary?

**Review by mreso — APPROVED** [review]
Left some comments. The readme needs some spell checking. Only major issue is probably the path for the results.json (see comment)

File: eval/open_llm_leaderboard/hellaswag_utils.py (hunk 2 above)
Comment (HamidShojanazeri) [reply eval/open_llm_leaderboard/hellaswag_utils.py:1]: yes, it does not pick it up from there even though use `include`.

File: eval/eval.py [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+            gen_kwargs=args.gen_kwargs,
+        )
+        handle_output(args, results, logger)
+
+    except Exception as e:
+        logger.error(f"An error occurred during evaluation: {e}")
+        sys.exit(1)
+
+
+if __name__ == "__main__":
```
[the remark this answers is not in the document]
Comment (HamidShojanazeri) [reply eval/eval.py:226]: one main issue is we need to clone their repo, as would need their yaml file path for extension of those config file for llm-leaderboard.

**Pull request merged**